test/CMakeLists.txt

```diff
@@ -79,15 +81,19 @@ set(GRAPHICS_SRC
     Graphics/Shape.test.cpp
     Graphics/Sprite.test.cpp
     Graphics/Text.test.cpp
-    Graphics/Texture.test.cpp
     Graphics/Transform.test.cpp
     Graphics/Transformable.test.cpp
     Graphics/Vertex.test.cpp
     Graphics/VertexArray.test.cpp
     Graphics/VertexBuffer.test.cpp
     Graphics/View.test.cpp
 )
-sfml_add_test(test-sfml-graphics "${GRAPHICS_SRC}" SFML::Graphics)
+if(SFML_BUILD_OPENGL_TESTS)
+    set(EXTRA_GRAPHICS_SRC
+        Graphics/Texture.test.cpp
```

Comment:
I guess there's a discussion point of whether we'd need dedicated classes, as the asserts could for example be executed on non-graphical instances as well.

Reply:
Instead of selectively building different .cpp files we could have
```cmake
if(SFML_BUILD_DISPLAY_TESTS)
    target_compile_definitions(sfml-test-graphics PRIVATE SFML_BUILD_DISPLAY_TESTS)
endif()
```
and
```cpp
#if SFML_BUILD_DISPLAY_TESTS
// All the stuff that requires a display
#endif
```
That would ensure that all headers and their respective `static_asserts` always get built. I'm on the fence about whether this is worth it.

Reply:
This was not hard to implement so I switched to a different method which ensures these .cpp files always get built.